### PR TITLE
Add first iteration json-stat SSB-pocos

### PR DESCRIPTION
**Pull request by Kpaubert**
Updates #44

### CODE REVIEW

File: Models/JsonStat/JsonStatGenerics.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Text.Json.Serialization;
+
+namespace Backend.Models.JsonStat
+{
+    public abstract class JsonStatGenericDataset
```
Comment (Kpaubert) [inline Models/JsonStat/JsonStatGenerics.cs:8]: Perhaps rename this entire file, as ["generics" is reserved for other purposes](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/).

File: Models/JsonStat/SSBAgesInNorway.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+using System.Text.Json.Serialization;
+
+namespace Backend.Models.JsonStat
+{
+    // Remove comments when merge-ready
```
Comment (Kpaubert) [inline Models/JsonStat/SSBAgesInNorway.cs:5]: Just another reminder.

File: Models/JsonStat/SSBAgesInNorway.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    // Based on https://data.ssb.no/api/v0/dataset/1074.json?lang=no
+    public class SSBAgesInNorway
```
Comment (Kpaubert) [inline Models/JsonStat/SSBAgesInNorway.cs:7]: Can we abstract even further?

**Comment by Kpaubert** [comment]
Another topic for tomorrow: directory setup for more reusable classes (i.e. formats). We could perhaps create another root directory containing the json-stat and other base-classes that can be implemented in the models directory by other POCO's.

Ex:
BaseModels
- JsonStat
  - JsonStatBase.cs

Models
- MetAPI
  - _all MetAPI-stuff (sidenote: does this conform to any widely used standard? We could perhaps abstract this one as well)_
- SSB
  - AgesInNorway.cs _(which implements JsonStatBase.cs)_

This way we don't need json-stat-files in all directories containing json-stat-like objects.

File: Models/JsonStat/JsonStatGenerics.cs (hunk 1 above)
Comment (FredrikAugust) [reply Models/JsonStat/JsonStatGenerics.cs:8]: Agreed.

Options:
- Remove `Generic`
- Use `Base` instead

File: Models/JsonStat/JsonStatGenerics.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        [JsonPropertyName("label")] public string Label { get; set; }
+        [JsonPropertyName("source")] public string Source { get; set; }
+        [JsonPropertyName("updated")] public DateTime Updated { get; set; }
+        [JsonPropertyName("value")] public Collection<int> Value { get; set; }
+    }
+
+    public abstract class JsonStatGenericDimension
```
Comment (FredrikAugust) [inline Models/JsonStat/JsonStatGenerics.cs:16]: ^

File: Models/JsonStat/SSBAgesInNorway.cs (hunk 2 above)
Comment (FredrikAugust) [reply Models/JsonStat/SSBAgesInNorway.cs:5]: In JetBrains you can add `TODO: ` to have it show up in the TODO pane (though the comments feel uncomfortable in `master`).

File: Models/JsonStat/SSBAgesInNorway.cs (hunk 3 above)
Comment (FredrikAugust) [reply Models/JsonStat/SSBAgesInNorway.cs:7]: I believe you can make the `class` generic, and make the `Dimension` property use this generic type.

File: Models/JsonStat/SSBAgesInNorway.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        /*
+         * I tried abstracting this class as much as possible, so that the only thing
+         * that has to be changed when creating a new JSON-stat object, only the dimension
+         * needs to be changed.
+         */
```
Comment (FredrikAugust) [inline Models/JsonStat/SSBAgesInNorway.cs:13]: I believe you can use `///` to create MSDocs (I think that's what they are called) to provide other developers with extra information regarding a class/property/function/&c.

Comment (Kpaubert) [reply Models/JsonStat/SSBAgesInNorway.cs:13]: Indeed, but that's mainly used for comments that should be in the code, even after merging.
E.g.
```cs
/// <summary>
/// Squares the input integer
/// </summary>
/// <param name="x">Input integer</param>
/// <returns>The square of the input</returns>
public int Square(int x)
{
    return x * x;
}
```

Comment (FredrikAugust) [reply Models/JsonStat/SSBAgesInNorway.cs:13]: Fair point! Thought these were supposed to be in `master`. _After rebase_

File: Models/SSB/AgesInNorway.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        [JsonPropertyName("dataset")] public JsonStatDataset<AgesInNorwayDimension> Dataset { get; set; }
+
+        public sealed class AgesInNorwayDimension : AbstractJsonStatDimension
+        {
+            [JsonPropertyName("Region")] public JsonStatDimensionContent Region { get; set; }
+            [JsonPropertyName("Tid")] public JsonStatDimensionContent Tid { get; set; }
+        }
+    }
+}
```
Comment (blauks) [inline Models/SSB/AgesInNorway.cs:16]: EOL

**Review by fredrikbw — APPROVED** [review]
Nicely done

**Pull request merged**